ingestion/src/metadata/utils/datalake/datalake_utils.py

```diff
@@ -472,3 +486,20 @@ def _get_pq_data_type(self, column):
             data_type = self._data_formats.get(str(column.type), DataType.UNKNOWN)
 
         return data_type
+
+
+# pylint: disable=import-outside-toplevel
+class JsonDataFrameColumnParser(GenericDataFrameColumnParser):
+    """Given a dataframe object generated from a json file, parse the columns and return a list of Column objects."""
+
+    def get_columns(self):
+        """
+        method to process column details for json files
+        """
+        if self.raw_data:
+            try:
+                return parse_json_schema(schema_text=self.raw_data, cls=Column)
+            except Exception as exc:
```

Comment:
instead of try catch, isn't there a "viable" approach for validating if a JSON file is a JSON Schema?

Reply:
@pmbrull we are only parsing json schema when we have `$schema`, this try catch is just to handle scenario when something went wrong while processing the json schema itself.